src/renderer/components/+network-endpoints/endpoints.store.ts

```diff
@@ -7,9 +7,15 @@ import { KubeObjectStore } from "../../../common/k8s-api/kube-object.store";
 import type { Endpoints, EndpointsApi, EndpointsData } from "../../../common/k8s-api/endpoints/endpoint.api";
 import { endpointsApi } from "../../../common/k8s-api/endpoints/endpoint.api";
 import { apiManager } from "../../../common/k8s-api/api-manager";
+import { isClusterPageContext } from "../../utils";
 
 export class EndpointStore extends KubeObjectStore<Endpoints, EndpointsApi, EndpointsData> {
 }
 
-export const endpointStore = new EndpointStore(endpointsApi);
-apiManager.registerStore(endpointStore);
+export const endpointStore = isClusterPageContext()
+  ? new EndpointStore(endpointsApi)
+  : undefined as never;
+
+if (isClusterPageContext()) {
+  apiManager.registerStore(endpointStore);
+}
```

Comment:
Big nit: There's a lot wrong here:

1. Import-time side-effect
2. Import-time logic
3. `isClusterPageContext`, what does that mean?
4. Intent of code is unclear